tech_demos/loader_animations/pure_css_stars_bevel.html

```diff
@@ -58,57 +66,147 @@
-        clearInterval(anim_timer);
+      current_percent = global_percent;
+    }
+  }
+  else if (current_percent >= 100) {
+    // Wait until the stars have all finished before we animate the flare!
+    for (var id in star_animation_timing) {
+      var star = svg.getElementById(id);
+      if (star.style.animationPlayState == 'running') {
+//	console.max("oi!", "waiting for stars to finish before doing flare");
+//	return;
       }
     }
+//    console.max("hey!", "stars are all done, running flare");
+
+    svg.getElementById('flare_top').style.display = 'block';
+    svg.getElementById('flare_bottom').style.display = 'block';
+    clearInterval(anim_timer);
+  }
+
+  // already requested a percent before we were fully loaded, kick things off for real
+  if (!anim_timer && global_percent > 0) {
+    setLoaderPercent(global_percent);
+  }
+}
 
-    anim_timer = setInterval(animation_cycler, 100);
 
+$('#logo_svg').ready(function() {
+  loader_ready = true;
+});
 
-    setInterval(function(){
-        global_percent += 5;
+$(document).ready(function() {
+  $('#pos_slider').slider({
+    min: 0,
+    max: 100,
+    step: 1,
+    slide: function (e, ui) {
+      if (ui.value < global_percent) {
+	e.preventDefault();
+      }
+      else {
+	if (global_percent != ui.value) {
+	  setLoaderPercent(ui.value);
+	}
+      }
+    }
+  });
+});
+
+function setLoaderPercent(loader_percent)
+{
+  if (loader_percent > 100) {
+    loader_percent = 100;
+  }
+  global_percent = loader_percent;
+
+  if (loader_ready) {
+    if (!anim_timer) {
+      anim_timer = setInterval(animation_cycler, 100);
+    }
 
-        if (global_percent > 100) {
-          global_percent = 100;
-        }
+    $('#percentage').html(global_percent);
+    $('#pos_slider').slider('value', global_percent);
+  }
+}
 
-        $('#percentage').html(global_percent);
-      }, 50);
-    });
+function testReset()
+{
+  loader_ready = false
+  current_percent = 0;
+  global_percent = 0;
+  $('#percentage').html(0);
+  $('#pos_slider').slider('value', 0);
+  $('#logo_svg').attr('data', 'ytmnd_logo_bevel.svg');
+    
+  $('#logo_svg').ready(function() {
+    loader_ready = true;
   });
+}
 
+
+var autoInterval = null;
+function testAuto(start, end, duration)
+{
+  if (autoInterval) {
+    clearInterval(autoInterval)
+  }
+
+  var interval = (duration) / (end - start);
+  setLoaderPercent(start);
+  
+  autoInterval = setInterval(function(){
+    setLoaderPercent(global_percent + 1);
+    if (global_percent >= end) {
+      clearInterval(autoInterval);
+    }
+  }, interval);
+}
+  
 </script>
+
+<Style type="text/css">
+  .ui-draggable, .ui-droppable {
+  background-position: top;
+  }</style>
 </head>
 <body>
 
+  <div>
 <div id="ytmnd_logo_div">
-  <object id="logo_svg" type="image/svg+xml" data="ytmnd_logo_bevel.svg"></object>
+  <object id="logo_svg" type="image/svg+xml" data="ytmnd_logo_bevel.svg" style="width: 819px; height: 531px;"></object>
```

Comment:
Is this necessary? This will need to be dynamically sized based on the viewport size.

Reply:
It's necessary in the context that it's being used at the moment, as the object can and does end up with non-integer dimensions. Assuming the code that puts it on the screen can do this properly, it's not needed, but in this static harness it is. Also fixing the canvas dimensions inside the SVG might help, but I haven't looked into that yet.

Reply:
Okay thanks. What were the effects of it having non-integer dimensions? Is that the frame issue that rachelfish noticed?

Reply:
Yes. Potential aliasing including the border issue. Screens work in pixels.